compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/ConstLowering.kt

```diff
@@ -30,7 +30,7 @@ fun IrField.constantValue(implicitConst: Boolean = false): IrConst<*>? {
     // Kotlin fields are only inlined if explicitly `const` to avoid making the values part of the ABI.
     val inline = correspondingPropertySymbol?.owner?.isConst == true ||
             (isStatic && isFinal && (origin == IrDeclarationOrigin.IR_EXTERNAL_JAVA_DECLARATION_STUB || implicitConst))
-    return if (inline) initializer?.expression as? IrConst<*> else null
+    return if (inline) (initializer?.expression as? IrConst<*>)?.copy() else null
```

Comment:
Is it required for ir-validation? Or is it another reason?

Reply:
Yeah, that's for validation.

>NB: maybe it's worth to rewrite using #2610 approach?

Not sure what you mean, as in that PR the task is to locate a property by name while here we already have the property/field.

Reply:
Than my question doesn't matter

Reply:
Rolled back to avoid conflicts with #2640